### DISCUSSION TITLE
Canonical smiles generation

### DISCUSSION THREAD

**Discussion by sara-hashemi** [rdkit/rdkit#7842] [General]
I am using the rdkit's Chem library and importing AllChem to generate canonical smiles from inchi, InChIKey and PC_CID. When comparing the generated canonical smiles based on inchi and PubChem CID, the smiles don't match. I would like to see why the smiles generated through inchi do not match with those generated by pubchem CID or InChIKey.

**Comment by jasondbiggs** [comment]
How do you make a SMILES string from an InChIKey or CID number?

**Reply by sara-hashemi** [reply]
Both are based on pubchempy. I use the below function to get the smiles string from CID:

import pubchempy as pcp
def get_smiles_from_cid(cid):
    try:
        compound = pcp.Compound.from_cid(cid)
        return compound.canonical_smiles if compound else None
    except Exception as e:
        print(f"Error fetching PubChem SMILES for CID {cid}: {e}")
    return None

And the below code to generate the canonical smiles from InChIKey:

import pubchempy as pcp
def get_smiles_from_inchikey(inchikey):
    try:
        compound = pcp.get_compounds(inchikey, 'inchikey')
        if compound:
            return compound[0].canonical_smiles
    except Exception as e:
        print(f"Error fetching PubChem SMILES for InChIKey {inchikey}: {e}")
    return None

**Comment by sara-hashemi** [comment]
I use the below function to get the smiles string from CID:

import pubchempy as pcp
def get_smiles_from_cid(cid):
    try:
        compound = pcp.Compound.from_cid(cid)
        return compound.canonical_smiles if compound else None
    except Exception as e:
        print(f"Error fetching PubChem SMILES for CID {cid}: {e}")
    return None

And the below code to generate the canonical smiles from InChIKey:

import pubchempy as pcp
def get_smiles_from_inchikey(inchikey):
    try:
        compound = pcp.get_compounds(inchikey, 'inchikey')
        if compound:
            return compound[0].canonical_smiles
    except Exception as e:
        print(f"Error fetching PubChem SMILES for InChIKey {inchikey}: {e}")
    return None

**Comment by bp-kelley** [comment]
The short answer is that pubchem's canonical smiles were created via the OpenEye toolkit ( a closed source commercial toolkit ) and there is no expectation that different canonicalization algorithms generate the same smiles.  There is no "one standard" canonicalization, each toolkit has a different implementation.

In general, if you are using your own database or records, you need to choose a canonicalization that you can reproduce.

```
>>> import pubchempy as pcp
>>> from rdkit import Chem
>>> cid = 5381226
>>> compound = pcp.Compound.from_cid(cid)
>>> compound.isomeric_smiles
'C[C@H]1/C=C/C=C(\\C(=O)NC\\2=C(C3=C(C4=C(C(=C3O)C)O[C@@](C4=O)(O/C=C/[C@@H]([C@H]([C@H]([C@@H]([C@@H]([C@@H]([C@H]1O)C)O)C)OC(=O)C)C)OC)C)C(=O)/C2=C/NN5CCN(CC5)C)O)/C'
>>> Chem.CanonSmiles(compound.isomeric_smiles) == compound.isomeric_smiles
False
```
